send-transaction-service/src/send_transaction_service.rs

```diff
@@ -410,7 +441,8 @@ impl SendTransactionService {
                 let now = Instant::now();
                 let expired = transaction_info
                     .last_sent_time
-                    .map(|last| now.duration_since(last) >= retry_rate)
+                    .and_then(|last| now.checked_duration_since(last))
+                    .map(|elapsed| elapsed >= retry_rate)
                     .unwrap_or(false);
```

Comment:
if `checked_duration_since()` returns `None` we are likely expired, yet we still fallback to "has never been sent"